src/lib/comms/sol-oic-client.c

```diff
@@ -56,6 +57,12 @@
 #define OIC_RESOURCE_CHECK_API(ptr, ...)
 #endif
 
+#define CHECK_REQUEST(request, ...) \
```

Comment:
several public functions are calling this macro directly without check if `request` is `NULL`. You can add this check in this macro to avoid call `SOL_NULL_CHECK` before.

If you not want to do this, review where this macro is being used and check whether the null check  is necessary.